Update `new`, `help`, output styles

joshuayoes opened the pull request:
## Please verify the following:

- [x] `yarn ci:test` **jest** tests pass with new tests, if relevant
- [x] `README.md` has been updated with your changes, if relevant

## Describe your PR
Resolves https://github.com/infinitered/ignite/issues/2114

joshuayoes commented:
Here is a draft of the direction I'm going
- add ignite ascii art to the beginning
- replaced extra colors with bold emphasis to create a consistent color scheme
- added link styles to all links

I want to do some more work with the spinners, but I'm pretty happy with this so far

<img width="835" alt="ignite new output" src="https://user-images.githubusercontent.com/37849890/188250104-6728fc3d-a7ec-418a-947b-ced057f45cb4.png">

jamonholmgren commented:
Looking pretty good! I miss some of the colors a bit -- a lot of CLIs use monochrome, and I like that ours is more colorful. At least the red for `ignite-cli` and `Infinite Red`.

joshuayoes commented:
Add npm, yarn, pnpm brand colors to output

joshuayoes commented:
Add more decoration to prompts

joshuayoes commented:
Added unique colors for yarn, npm, and pnpm

<img width="751" alt="Screen Shot 2022-09-07 at 11 25 18 AM" src="https://user-images.githubusercontent.com/37849890/188951333-dcdc71e9-5cf8-4f05-880d-1a16d22f63b2.png">
<img width="757" alt="Screen Shot 2022-09-07 at 11 25 09 AM" src="https://user-images.githubusercontent.com/37849890/188951340-eca2b4f7-cced-49b4-a8af-36cca3b300c4.png">
<img width="747" alt="Screen Shot 2022-09-07 at 11 24 54 AM" src="https://user-images.githubusercontent.com/37849890/188951355-788cd5f4-4963-4640-9160-a731b0c3c21d.png">

joshuayoes commented:
Added red background and white bold text to Infinite Red, along with Infinity Emoji to make it pop

<img width="925" alt="Screen Shot 2022-09-07 at 6 48 59 PM" src="https://user-images.githubusercontent.com/37849890/189015587-e13c003a-7678-4401-9f13-739f42965dc8.png">

joshuayoes commented:
https://user-images.githubusercontent.com/37849890/189021962-314d32e4-c364-4add-bc77-0fb7c6236a70.mov

Figured out how to replace prompt prefixs with emojis 👀

joshuayoes commented:
Boolean options now have emojis, and also display 'Yes' or 'No' instead of 'true' or 'false'

https://user-images.githubusercontent.com/37849890/189189649-9d3a20d4-f1cf-42d5-b8d8-057a405c0c8c.mov

joshuayoes commented:
Added some more humor to the spinner messages

Fresh install

<img width="401" alt="Screen Shot 2022-09-08 at 3 57 04 PM" src="https://user-images.githubusercontent.com/37849890/189240131-e8a3b5d5-0b41-4d62-8f7e-07399351c12d.png">

Cached dependencies

<img width="375" alt="Screen Shot 2022-09-08 at 3 57 18 PM" src="https://user-images.githubusercontent.com/37849890/189240148-69481a09-f9e2-4841-b32a-a7a39948a78e.png">

joshuayoes commented:
More style changes:
- Added highlights for app name, 
- Added IR highlight to Ignite version
- Changed packager colors to bg highlights

<img width="889" alt="Screen Shot 2022-09-08 at 3 59 46 PM" src="https://user-images.githubusercontent.com/37849890/189240405-5ae01d73-f45d-44f0-a2d5-ac2a0c660ae9.png">

jamonholmgren commented on src/commands/new.ts:
```diff
@@ -152,8 +162,6 @@ export default {
     // #endregion
 
     // #region Project Name
-    heading("🔥 About to Ignite your new app! 🔥")
```
Could we add two blank lines before the initial question? It feels weird being smashed up against the command prompt.

```ts
p()
p()
```

joshuayoes replied:
Yeah I can add that!

jamonholmgren replied:
I like visual space -- as you can tell :D

joshuayoes replied:
<img width="464" alt="Screen Shot 2022-09-09 at 10 32 40 AM" src="https://user-images.githubusercontent.com/37849890/189410657-100f8d5c-07d1-4ccf-a7b5-d47bcf06eef4.png">

[feat(new): add a line break before question prompts](https://github.com/infinitered/ignite/pull/2119/commits/15a89a7c0ea541b4a66eee6d30f2d01c05497036)

jamonholmgren commented on src/commands/new.ts:
```diff
@@ -503,116 +544,155 @@ export default {
+  const privateFlags: Flag[] = [
+    "b",
+    "boilerplate",
+    "coloLoco",
```
Not now, but before our full launch, let's dump any references to coloLoco in the CLI.

joshuayoes replied:
This was pretty straightforward, so I threw it in here:

[feat(new): remove references to colo-loco](https://github.com/infinitered/ignite/pull/2119/commits/d294be6998c09167492e37ae7b9ef03408a4756b)

jamonholmgren commented on test/vanilla/ignite-new.test.ts:
```diff
@@ -42,8 +42,8 @@ describe("ignite new", () => {
 
     it("should print expected text to std out", () => {
-      expect(result).toContain(`Using ignite-cli`)
-      expect(result).toContain(`Ignite CLI ignited ${APP_NAME}`)
```
I think this line, or something equivalent, should be present, as it'll fail if Ignite fails to fully spin up an app and print the success messages.

joshuayoes replied:
Ah! I knew that it was printing std out, but for some reason it did not click that we were looking for the success message

joshuayoes replied:
Added here: [feat(ignite-new): add success message test](https://github.com/infinitered/ignite/pull/2119/commits/b5a16064861a24977523b5690ad20ac8c7400886)

jamonholmgren reviewed (approved):
2 minor nits, but otherwise I think we're ready to merge!
